src/services/publicCollectionDownloadManager.ts

```diff
@@ -113,15 +117,15 @@ class PublicCollectionDownloadManager {
                 );
                 const fileBlob = await new Response(fileStream).blob();
                 if (forPreview) {
-                    const convertedBlobs = await convertForPreview(
-                        file,
-                        fileBlob
-                    );
-                    return convertedBlobs.map((blob) =>
-                        URL.createObjectURL(blob)
-                    );
+                    return await getRenderableFileURL(file, fileBlob);
+                } else {
+                    return [
```

Comment:
Let's wrap this in an object instead of returning an array?

Reply:
That will touch many more code paths than I want to change right now.

Currently, I refactored the function to perform a single task, which was the main issue.

using object vs array is more of a design choice. 

Planning to  do it in a separate PR later on